src/contracts/fungible/cache/sql.rs

```diff
@@ -264,6 +264,86 @@ impl Cache for SqlCache {
         self.save()?;
         Ok(existed)
     }
+
+    fn utxo_allocation_map(
+        &self,
+        contract_id: ContractId,
+    ) -> Result<HashMap<bitcoin::OutPoint, Vec<AtomicValue>>, CacheError> {
+        // Get the required asset from cache
+        let asset = self.asset(contract_id)?;
+
+        // Fetch its known_allocation map
+        let allocation_map = asset.known_allocations();
+
+        // Process known_allocation map to produce the intended map
+        let mut result = HashMap::new();
+
+        for item in allocation_map {
+            let mut alloc_amount = vec![];
+            for alloc in item.1 {
+                alloc_amount.push(alloc.value().value);
+            }
+            result.insert(item.0.clone(), alloc_amount);
+        }
+
+        Ok(result)
```

Comment:
Pls see sample from `FileCache`; the same applies here

Reply:
fixed